octopoes/tests/integration/test_xtdb_client.py

```diff
@@ -199,6 +202,18 @@ def test_query_where_in(xtdb_session: XTDBSession, valid_time: datetime):
 
     assert len(xtdb_session.client.query(query, valid_time)) == 0
 
+    object_path = Path.parse("IPAddress.network")
+    pk = A(IPAddress, field="primary_key")
+    query = (
+        Query.from_path(object_path)
+        .find(pk)
+        .pull(IPAddress)
+        .where(IPAddress, network=Network)
+        .where(IPAddress, primary_key=pk)
+        .where_in(IPAddress, primary_key=["IPAddressV4|testnetwork2|127.0.0.1", "IPAddressV4|testnetwork|0.0.0.0"])
+    )
+    assert len(xtdb_session.client.query(query, valid_time)) == 1
```

Comment:
Please validate output against the queried set (`[ IPAddressV4 ]`), since validating the number of output objects isn't strong enough. For example, what if the query indeed returned one object as requested, but the wrong one?